diff --git a/Cargo.toml b/Cargo.toml
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -19,50 +19,39 @@
 version = "1"

 [dependencies.aws-credential-types]
-path = "../aws-credential-types"
 version = "1.3.0"

 [dependencies.aws-runtime]
-path = "../aws-runtime"
 version = "1.9.2"

 [dependencies.aws-smithy-async]
-path = "../aws-smithy-async"
 version = "1.3.0"

 [dependencies.aws-smithy-http]
-path = "../aws-smithy-http"
 version = "0.64.0"

 [dependencies.aws-smithy-json]
-path = "../aws-smithy-json"
 version = "0.63.0"

 [dependencies.aws-smithy-observability]
-path = "../aws-smithy-observability"
 version = "0.3.0"

 [dependencies.aws-smithy-runtime]
-path = "../aws-smithy-runtime"
 features = ["client"]
 version = "1.14.0"

 [dependencies.aws-smithy-runtime-api]
-path = "../aws-smithy-runtime-api"
 features = ["client", "http-1x"]
 version = "1.16.0"

 [dependencies.aws-smithy-schema]
-path = "../aws-smithy-schema"
 version = "0.2.0"

 [dependencies.aws-smithy-types]
-path = "../aws-smithy-types"
 features = ["http-body-1-x"]
 version = "1.6.3"

 [dependencies.aws-types]
-path = "../aws-types"
 version = "1.6.0"

 [dependencies.bytes]
@@ -84,7 +73,6 @@
 [dependencies.tracing]
 version = "0.1"
 [dev-dependencies.aws-credential-types]
-path = "../aws-credential-types"
 features = ["test-util"]
 version = "1.3.0"

@@ -103,3 +91,7 @@
 test-util = ["aws-credential-types/test-util", "aws-smithy-runtime/legacy-test-util", "aws-smithy-runtime/test-util"]
 gated-tests = []
 default = ["rustls", "default-https-client", "rt-tokio"]
+op_create_token = []
+op_create_token_with_iam = []
+op_register_client = []
+op_start_device_authorization = []
diff --git a/src/client.rs b/src/client.rs
--- a/src/client.rs
+++ b/src/client.rs
@@ -90,8 +90,10 @@
     }
 }

+#[cfg(feature = "op_create_token")]
 mod create_token;

+#[cfg(feature = "op_create_token_with_iam")]
 mod create_token_with_iam;

 /// Operation customization and supporting types.
@@ -121,6 +123,8 @@
 /// ```
 pub mod customize;

+#[cfg(feature = "op_register_client")]
 mod register_client;

+#[cfg(feature = "op_start_device_authorization")]
 mod start_device_authorization;
diff --git a/src/error_meta.rs b/src/error_meta.rs
--- a/src/error_meta.rs
+++ b/src/error_meta.rs
@@ -3,32 +3,77 @@
 #[non_exhaustive]
 #[derive(::std::fmt::Debug)]
 pub enum Error {
+    #[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
     /// <p>You do not have sufficient access to perform this action.</p>
     AccessDeniedException(crate::types::error::AccessDeniedException),
+    #[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
     /// <p>Indicates that a request to authorize a client with an access user session token is pending.</p>
     AuthorizationPendingException(crate::types::error::AuthorizationPendingException),
+    #[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
     /// <p>Indicates that the token issued by the service is expired and is no longer valid.</p>
     ExpiredTokenException(crate::types::error::ExpiredTokenException),
+    #[cfg(any(
+        feature = "op_create_token",
+        feature = "op_create_token_with_iam",
+        feature = "op_register_client",
+        feature = "op_start_device_authorization"
+    ))]
     /// <p>Indicates that an error from the service occurred while trying to process a request.</p>
     InternalServerException(crate::types::error::InternalServerException),
+    #[cfg(any(
+        feature = "op_create_token",
+        feature = "op_create_token_with_iam",
+        feature = "op_start_device_authorization"
+    ))]
     /// <p>Indicates that the <code>clientId</code> or <code>clientSecret</code> in the request is invalid. For example, this can occur when a client sends an incorrect <code>clientId</code> or an expired <code>clientSecret</code>.</p>
     InvalidClientException(crate::types::error::InvalidClientException),
+    #[cfg(feature = "op_register_client")]
     /// <p>Indicates that the client information sent in the request during registration is invalid.</p>
     InvalidClientMetadataException(crate::types::error::InvalidClientMetadataException),
+    #[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
     /// <p>Indicates that a request contains an invalid grant. This can occur if a client makes a <code>CreateToken</code> request with an invalid grant type.</p>
     InvalidGrantException(crate::types::error::InvalidGrantException),
+    #[cfg(feature = "op_register_client")]
     /// <p>Indicates that one or more redirect URI in the request is not supported for this operation.</p>
     InvalidRedirectUriException(crate::types::error::InvalidRedirectUriException),
+    #[cfg(any(
+        feature = "op_create_token",
+        feature = "op_create_token_with_iam",
+        feature = "op_register_client",
+        feature = "op_start_device_authorization"
+    ))]
     /// <p>Indicates that something is wrong with the input to the request. For example, a required parameter might be missing or out of range.</p>
     InvalidRequestException(crate::types::error::InvalidRequestException),
+    #[cfg(feature = "op_create_token_with_iam")]
     /// <p>Indicates that a token provided as input to the request was issued by and is only usable by calling IAM Identity Center endpoints in another region.</p>
     InvalidRequestRegionException(crate::types::error::InvalidRequestRegionException),
+    #[cfg(any(
+        feature = "op_create_token",
+        feature = "op_create_token_with_iam",
+        feature = "op_register_client"
+    ))]
     /// <p>Indicates that the scope provided in the request is invalid.</p>
     InvalidScopeException(crate::types::error::InvalidScopeException),
+    #[cfg(any(
+        feature = "op_create_token",
+        feature = "op_create_token_with_iam",
+        feature = "op_register_client",
+        feature = "op_start_device_authorization"
+    ))]
     /// <p>Indicates that the client is making the request too frequently and is more than the service can handle.</p>
     SlowDownException(crate::types::error::SlowDownException),
+    #[cfg(any(
+        feature = "op_create_token",
+        feature = "op_create_token_with_iam",
+        feature = "op_start_device_authorization"
+    ))]
     /// <p>Indicates that the client is not currently authorized to make the request. This can happen when a <code>clientId</code> is not issued for a public client.</p>
     UnauthorizedClientException(crate::types::error::UnauthorizedClientException),
+    #[cfg(any(
+        feature = "op_create_token",
+        feature = "op_create_token_with_iam",
+        feature = "op_register_client"
+    ))]
     /// <p>Indicates that the grant type in the request is not supported by the service.</p>
     UnsupportedGrantTypeException(crate::types::error::UnsupportedGrantTypeException),
     /// An unexpected error occurred (e.g., invalid JSON returned by the service or an unknown error code).
@@ -45,19 +90,64 @@
 impl ::std::fmt::Display for Error {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         match self {
+            #[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
             Error::AccessDeniedException(inner) => inner.fmt(f),
+            #[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
             Error::AuthorizationPendingException(inner) => inner.fmt(f),
+            #[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
             Error::ExpiredTokenException(inner) => inner.fmt(f),
+            #[cfg(any(
+                feature = "op_create_token",
+                feature = "op_create_token_with_iam",
+                feature = "op_register_client",
+                feature = "op_start_device_authorization"
+            ))]
             Error::InternalServerException(inner) => inner.fmt(f),
+            #[cfg(any(
+                feature = "op_create_token",
+                feature = "op_create_token_with_iam",
+                feature = "op_start_device_authorization"
+            ))]
             Error::InvalidClientException(inner) => inner.fmt(f),
+            #[cfg(feature = "op_register_client")]
             Error::InvalidClientMetadataException(inner) => inner.fmt(f),
+            #[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
             Error::InvalidGrantException(inner) => inner.fmt(f),
+            #[cfg(feature = "op_register_client")]
             Error::InvalidRedirectUriException(inner) => inner.fmt(f),
+            #[cfg(any(
+                feature = "op_create_token",
+                feature = "op_create_token_with_iam",
+                feature = "op_register_client",
+                feature = "op_start_device_authorization"
+            ))]
             Error::InvalidRequestException(inner) => inner.fmt(f),
+            #[cfg(feature = "op_create_token_with_iam")]
             Error::InvalidRequestRegionException(inner) => inner.fmt(f),
+            #[cfg(any(
+                feature = "op_create_token",
+                feature = "op_create_token_with_iam",
+                feature = "op_register_client"
+            ))]
             Error::InvalidScopeException(inner) => inner.fmt(f),
+            #[cfg(any(
+                feature = "op_create_token",
+                feature = "op_create_token_with_iam",
+                feature = "op_register_client",
+                feature = "op_start_device_authorization"
+            ))]
             Error::SlowDownException(inner) => inner.fmt(f),
+            #[cfg(any(
+                feature = "op_create_token",
+                feature = "op_create_token_with_iam",
+                feature = "op_start_device_authorization"
+            ))]
             Error::UnauthorizedClientException(inner) => inner.fmt(f),
+            #[cfg(any(
+                feature = "op_create_token",
+                feature = "op_create_token_with_iam",
+                feature = "op_register_client"
+            ))]
             Error::UnsupportedGrantTypeException(inner) => inner.fmt(f),
             Error::Unhandled(_) => {
                 if let ::std::option::Option::Some(code) =
@@ -82,24 +172,70 @@
 impl ::aws_smithy_types::error::metadata::ProvideErrorMetadata for Error {
     fn meta(&self) -> &::aws_smithy_types::error::metadata::ErrorMetadata {
         match self {
+            #[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
             Self::AccessDeniedException(inner) => inner.meta(),
+            #[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
             Self::AuthorizationPendingException(inner) => inner.meta(),
+            #[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
             Self::ExpiredTokenException(inner) => inner.meta(),
+            #[cfg(any(
+                feature = "op_create_token",
+                feature = "op_create_token_with_iam",
+                feature = "op_register_client",
+                feature = "op_start_device_authorization"
+            ))]
             Self::InternalServerException(inner) => inner.meta(),
+            #[cfg(any(
+                feature = "op_create_token",
+                feature = "op_create_token_with_iam",
+                feature = "op_start_device_authorization"
+            ))]
             Self::InvalidClientException(inner) => inner.meta(),
+            #[cfg(feature = "op_register_client")]
             Self::InvalidClientMetadataException(inner) => inner.meta(),
+            #[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
             Self::InvalidGrantException(inner) => inner.meta(),
+            #[cfg(feature = "op_register_client")]
             Self::InvalidRedirectUriException(inner) => inner.meta(),
+            #[cfg(any(
+                feature = "op_create_token",
+                feature = "op_create_token_with_iam",
+                feature = "op_register_client",
+                feature = "op_start_device_authorization"
+            ))]
             Self::InvalidRequestException(inner) => inner.meta(),
+            #[cfg(feature = "op_create_token_with_iam")]
             Self::InvalidRequestRegionException(inner) => inner.meta(),
+            #[cfg(any(
+                feature = "op_create_token",
+                feature = "op_create_token_with_iam",
+                feature = "op_register_client"
+            ))]
             Self::InvalidScopeException(inner) => inner.meta(),
+            #[cfg(any(
+                feature = "op_create_token",
+                feature = "op_create_token_with_iam",
+                feature = "op_register_client",
+                feature = "op_start_device_authorization"
+            ))]
             Self::SlowDownException(inner) => inner.meta(),
+            #[cfg(any(
+                feature = "op_create_token",
+                feature = "op_create_token_with_iam",
+                feature = "op_start_device_authorization"
+            ))]
             Self::UnauthorizedClientException(inner) => inner.meta(),
+            #[cfg(any(
+                feature = "op_create_token",
+                feature = "op_create_token_with_iam",
+                feature = "op_register_client"
+            ))]
             Self::UnsupportedGrantTypeException(inner) => inner.meta(),
             Self::Unhandled(inner) => &inner.meta,
         }
     }
 }
+#[cfg(feature = "op_create_token")]
 impl<R> From<::aws_smithy_runtime_api::client::result::SdkError<crate::operation::create_token::CreateTokenError, R>>
     for Error
 where
@@ -117,6 +253,7 @@
         }
     }
 }
+#[cfg(feature = "op_create_token")]
 impl From<crate::operation::create_token::CreateTokenError> for Error {
     fn from(err: crate::operation::create_token::CreateTokenError) -> Self {
         match err {
@@ -157,6 +294,7 @@
         }
     }
 }
+#[cfg(feature = "op_create_token_with_iam")]
 impl<R>
     From<
         ::aws_smithy_runtime_api::client::result::SdkError<
@@ -182,6 +320,7 @@
         }
     }
 }
+#[cfg(feature = "op_create_token_with_iam")]
 impl From<crate::operation::create_token_with_iam::CreateTokenWithIAMError> for Error {
     fn from(err: crate::operation::create_token_with_iam::CreateTokenWithIAMError) -> Self {
         match err {
@@ -227,6 +366,7 @@
         }
     }
 }
+#[cfg(feature = "op_register_client")]
 impl<R>
     From<::aws_smithy_runtime_api::client::result::SdkError<crate::operation::register_client::RegisterClientError, R>>
     for Error
@@ -248,6 +388,7 @@
         }
     }
 }
+#[cfg(feature = "op_register_client")]
 impl From<crate::operation::register_client::RegisterClientError> for Error {
     fn from(err: crate::operation::register_client::RegisterClientError) -> Self {
         match err {
@@ -276,6 +417,7 @@
         }
     }
 }
+#[cfg(feature = "op_start_device_authorization")]
 impl<R>
     From<
         ::aws_smithy_runtime_api::client::result::SdkError<
@@ -301,6 +443,7 @@
         }
     }
 }
+#[cfg(feature = "op_start_device_authorization")]
 impl From<crate::operation::start_device_authorization::StartDeviceAuthorizationError> for Error {
     fn from(err: crate::operation::start_device_authorization::StartDeviceAuthorizationError) -> Self {
         match err {
@@ -324,19 +467,64 @@
 impl ::std::error::Error for Error {
     fn source(&self) -> std::option::Option<&(dyn ::std::error::Error + 'static)> {
         match self {
+            #[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
             Error::AccessDeniedException(inner) => inner.source(),
+            #[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
             Error::AuthorizationPendingException(inner) => inner.source(),
+            #[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
             Error::ExpiredTokenException(inner) => inner.source(),
+            #[cfg(any(
+                feature = "op_create_token",
+                feature = "op_create_token_with_iam",
+                feature = "op_register_client",
+                feature = "op_start_device_authorization"
+            ))]
             Error::InternalServerException(inner) => inner.source(),
+            #[cfg(any(
+                feature = "op_create_token",
+                feature = "op_create_token_with_iam",
+                feature = "op_start_device_authorization"
+            ))]
             Error::InvalidClientException(inner) => inner.source(),
+            #[cfg(feature = "op_register_client")]
             Error::InvalidClientMetadataException(inner) => inner.source(),
+            #[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
             Error::InvalidGrantException(inner) => inner.source(),
+            #[cfg(feature = "op_register_client")]
             Error::InvalidRedirectUriException(inner) => inner.source(),
+            #[cfg(any(
+                feature = "op_create_token",
+                feature = "op_create_token_with_iam",
+                feature = "op_register_client",
+                feature = "op_start_device_authorization"
+            ))]
             Error::InvalidRequestException(inner) => inner.source(),
+            #[cfg(feature = "op_create_token_with_iam")]
             Error::InvalidRequestRegionException(inner) => inner.source(),
+            #[cfg(any(
+                feature = "op_create_token",
+                feature = "op_create_token_with_iam",
+                feature = "op_register_client"
+            ))]
             Error::InvalidScopeException(inner) => inner.source(),
+            #[cfg(any(
+                feature = "op_create_token",
+                feature = "op_create_token_with_iam",
+                feature = "op_register_client",
+                feature = "op_start_device_authorization"
+            ))]
             Error::SlowDownException(inner) => inner.source(),
+            #[cfg(any(
+                feature = "op_create_token",
+                feature = "op_create_token_with_iam",
+                feature = "op_start_device_authorization"
+            ))]
             Error::UnauthorizedClientException(inner) => inner.source(),
+            #[cfg(any(
+                feature = "op_create_token",
+                feature = "op_create_token_with_iam",
+                feature = "op_register_client"
+            ))]
             Error::UnsupportedGrantTypeException(inner) => inner.source(),
             Error::Unhandled(inner) => ::std::option::Option::Some(&*inner.source),
         }
@@ -345,19 +533,64 @@
 impl ::aws_types::request_id::RequestId for Error {
     fn request_id(&self) -> Option<&str> {
         match self {
+            #[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
             Self::AccessDeniedException(e) => e.request_id(),
+            #[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
             Self::AuthorizationPendingException(e) => e.request_id(),
+            #[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
             Self::ExpiredTokenException(e) => e.request_id(),
+            #[cfg(any(
+                feature = "op_create_token",
+                feature = "op_create_token_with_iam",
+                feature = "op_register_client",
+                feature = "op_start_device_authorization"
+            ))]
             Self::InternalServerException(e) => e.request_id(),
+            #[cfg(any(
+                feature = "op_create_token",
+                feature = "op_create_token_with_iam",
+                feature = "op_start_device_authorization"
+            ))]
             Self::InvalidClientException(e) => e.request_id(),
+            #[cfg(feature = "op_register_client")]
             Self::InvalidClientMetadataException(e) => e.request_id(),
+            #[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
             Self::InvalidGrantException(e) => e.request_id(),
+            #[cfg(feature = "op_register_client")]
             Self::InvalidRedirectUriException(e) => e.request_id(),
+            #[cfg(any(
+                feature = "op_create_token",
+                feature = "op_create_token_with_iam",
+                feature = "op_register_client",
+                feature = "op_start_device_authorization"
+            ))]
             Self::InvalidRequestException(e) => e.request_id(),
+            #[cfg(feature = "op_create_token_with_iam")]
             Self::InvalidRequestRegionException(e) => e.request_id(),
+            #[cfg(any(
+                feature = "op_create_token",
+                feature = "op_create_token_with_iam",
+                feature = "op_register_client"
+            ))]
             Self::InvalidScopeException(e) => e.request_id(),
+            #[cfg(any(
+                feature = "op_create_token",
+                feature = "op_create_token_with_iam",
+                feature = "op_register_client",
+                feature = "op_start_device_authorization"
+            ))]
             Self::SlowDownException(e) => e.request_id(),
+            #[cfg(any(
+                feature = "op_create_token",
+                feature = "op_create_token_with_iam",
+                feature = "op_start_device_authorization"
+            ))]
             Self::UnauthorizedClientException(e) => e.request_id(),
+            #[cfg(any(
+                feature = "op_create_token",
+                feature = "op_create_token_with_iam",
+                feature = "op_register_client"
+            ))]
             Self::UnsupportedGrantTypeException(e) => e.request_id(),
             Self::Unhandled(e) => e.meta.request_id(),
         }
diff --git a/src/operation.rs b/src/operation.rs
--- a/src/operation.rs
+++ b/src/operation.rs
@@ -1,14 +1,18 @@
 // Code generated by software.amazon.smithy.rust.codegen.smithy-rs. DO NOT EDIT.
 pub use ::aws_types::request_id::RequestId;

+#[cfg(feature = "op_create_token")]
 /// Types for the `CreateToken` operation.
 pub mod create_token;

+#[cfg(feature = "op_create_token_with_iam")]
 /// Types for the `CreateTokenWithIAM` operation.
 pub mod create_token_with_iam;

+#[cfg(feature = "op_register_client")]
 /// Types for the `RegisterClient` operation.
 pub mod register_client;

+#[cfg(feature = "op_start_device_authorization")]
 /// Types for the `StartDeviceAuthorization` operation.
 pub mod start_device_authorization;
diff --git a/src/protocol_serde.rs b/src/protocol_serde.rs
--- a/src/protocol_serde.rs
+++ b/src/protocol_serde.rs
@@ -1,4 +1,10 @@
 // Code generated by software.amazon.smithy.rust.codegen.smithy-rs. DO NOT EDIT.
+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_register_client",
+    feature = "op_start_device_authorization"
+))]
 pub(crate) fn type_erase_result<O, E>(
     result: ::std::result::Result<O, E>,
 ) -> ::std::result::Result<
@@ -17,6 +23,12 @@
         .map_err(::std::convert::Into::into)
 }

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_register_client",
+    feature = "op_start_device_authorization"
+))]
 pub fn parse_http_error_metadata(
     _response_status: u16,
     response_headers: &::aws_smithy_runtime_api::http::Headers,
@@ -28,14 +40,24 @@
     crate::json_errors::parse_error_metadata(response_body, response_headers)
 }

+#[cfg(feature = "op_create_token")]
 pub(crate) mod shape_create_token;

+#[cfg(feature = "op_create_token_with_iam")]
 pub(crate) mod shape_create_token_with_iam;

+#[cfg(feature = "op_register_client")]
 pub(crate) mod shape_register_client;

+#[cfg(feature = "op_start_device_authorization")]
 pub(crate) mod shape_start_device_authorization;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_register_client",
+    feature = "op_start_device_authorization"
+))]
 pub(crate) fn or_empty_doc(data: &[u8]) -> &[u8] {
     if data.is_empty() {
         b"{}"
@@ -44,42 +66,93 @@
     }
 }

+#[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
 pub(crate) mod shape_access_denied_exception;

+#[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
 pub(crate) mod shape_authorization_pending_exception;

+#[cfg(feature = "op_create_token")]
 pub(crate) mod shape_create_token_input;

+#[cfg(feature = "op_create_token_with_iam")]
 pub(crate) mod shape_create_token_with_iam_input;

+#[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
 pub(crate) mod shape_expired_token_exception;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_register_client",
+    feature = "op_start_device_authorization"
+))]
 pub(crate) mod shape_internal_server_exception;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_start_device_authorization"
+))]
 pub(crate) mod shape_invalid_client_exception;

+#[cfg(feature = "op_register_client")]
 pub(crate) mod shape_invalid_client_metadata_exception;

+#[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
 pub(crate) mod shape_invalid_grant_exception;

+#[cfg(feature = "op_register_client")]
 pub(crate) mod shape_invalid_redirect_uri_exception;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_register_client",
+    feature = "op_start_device_authorization"
+))]
 pub(crate) mod shape_invalid_request_exception;

+#[cfg(feature = "op_create_token_with_iam")]
 pub(crate) mod shape_invalid_request_region_exception;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_register_client"
+))]
 pub(crate) mod shape_invalid_scope_exception;

+#[cfg(feature = "op_register_client")]
 pub(crate) mod shape_register_client_input;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_register_client",
+    feature = "op_start_device_authorization"
+))]
 pub(crate) mod shape_slow_down_exception;

+#[cfg(feature = "op_start_device_authorization")]
 pub(crate) mod shape_start_device_authorization_input;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_start_device_authorization"
+))]
 pub(crate) mod shape_unauthorized_client_exception;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_register_client"
+))]
 pub(crate) mod shape_unsupported_grant_type_exception;

+#[cfg(feature = "op_create_token_with_iam")]
 pub(crate) mod shape_aws_additional_details;

+#[cfg(feature = "op_create_token_with_iam")]
 pub(crate) mod shape_scopes;
diff --git a/src/types.rs b/src/types.rs
--- a/src/types.rs
+++ b/src/types.rs
@@ -1,14 +1,30 @@
 // Code generated by software.amazon.smithy.rust.codegen.smithy-rs. DO NOT EDIT.
+#[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
 pub use crate::types::_access_denied_exception_reason::AccessDeniedExceptionReason;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_register_client",
+    feature = "op_start_device_authorization"
+))]
 pub use crate::types::_invalid_request_exception_reason::InvalidRequestExceptionReason;

+#[cfg(feature = "op_create_token_with_iam")]
 pub use crate::types::_aws_additional_details::AwsAdditionalDetails;

+#[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
 mod _access_denied_exception_reason;

+#[cfg(feature = "op_create_token_with_iam")]
 mod _aws_additional_details;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_register_client",
+    feature = "op_start_device_authorization"
+))]
 mod _invalid_request_exception_reason;

 /// Builders
diff --git a/src/types/builders.rs b/src/types/builders.rs
--- a/src/types/builders.rs
+++ b/src/types/builders.rs
@@ -1,2 +1,3 @@
 // Code generated by software.amazon.smithy.rust.codegen.smithy-rs. DO NOT EDIT.
+#[cfg(feature = "op_create_token_with_iam")]
 pub use crate::types::_aws_additional_details::AwsAdditionalDetailsBuilder;
diff --git a/src/types/error.rs b/src/types/error.rs
--- a/src/types/error.rs
+++ b/src/types/error.rs
@@ -1,58 +1,148 @@
 // Code generated by software.amazon.smithy.rust.codegen.smithy-rs. DO NOT EDIT.
+#[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
 pub use crate::types::error::_access_denied_exception::AccessDeniedException;

+#[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
 pub use crate::types::error::_authorization_pending_exception::AuthorizationPendingException;

+#[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
 pub use crate::types::error::_expired_token_exception::ExpiredTokenException;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_register_client",
+    feature = "op_start_device_authorization"
+))]
 pub use crate::types::error::_internal_server_exception::InternalServerException;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_start_device_authorization"
+))]
 pub use crate::types::error::_invalid_client_exception::InvalidClientException;

+#[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
 pub use crate::types::error::_invalid_grant_exception::InvalidGrantException;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_register_client",
+    feature = "op_start_device_authorization"
+))]
 pub use crate::types::error::_invalid_request_exception::InvalidRequestException;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_register_client"
+))]
 pub use crate::types::error::_invalid_scope_exception::InvalidScopeException;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_register_client",
+    feature = "op_start_device_authorization"
+))]
 pub use crate::types::error::_slow_down_exception::SlowDownException;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_start_device_authorization"
+))]
 pub use crate::types::error::_unauthorized_client_exception::UnauthorizedClientException;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_register_client"
+))]
 pub use crate::types::error::_unsupported_grant_type_exception::UnsupportedGrantTypeException;

+#[cfg(feature = "op_create_token_with_iam")]
 pub use crate::types::error::_invalid_request_region_exception::InvalidRequestRegionException;

+#[cfg(feature = "op_register_client")]
 pub use crate::types::error::_invalid_client_metadata_exception::InvalidClientMetadataException;

+#[cfg(feature = "op_register_client")]
 pub use crate::types::error::_invalid_redirect_uri_exception::InvalidRedirectUriException;

+#[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
 mod _access_denied_exception;

+#[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
 mod _authorization_pending_exception;

+#[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
 mod _expired_token_exception;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_register_client",
+    feature = "op_start_device_authorization"
+))]
 mod _internal_server_exception;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_start_device_authorization"
+))]
 mod _invalid_client_exception;

+#[cfg(feature = "op_register_client")]
 mod _invalid_client_metadata_exception;

+#[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
 mod _invalid_grant_exception;

+#[cfg(feature = "op_register_client")]
 mod _invalid_redirect_uri_exception;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_register_client",
+    feature = "op_start_device_authorization"
+))]
 mod _invalid_request_exception;

+#[cfg(feature = "op_create_token_with_iam")]
 mod _invalid_request_region_exception;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_register_client"
+))]
 mod _invalid_scope_exception;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_register_client",
+    feature = "op_start_device_authorization"
+))]
 mod _slow_down_exception;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_start_device_authorization"
+))]
 mod _unauthorized_client_exception;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_register_client"
+))]
 mod _unsupported_grant_type_exception;

 /// Builders
diff --git a/src/types/error/builders.rs b/src/types/error/builders.rs
--- a/src/types/error/builders.rs
+++ b/src/types/error/builders.rs
@@ -1,28 +1,73 @@
 // Code generated by software.amazon.smithy.rust.codegen.smithy-rs. DO NOT EDIT.
+#[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
 pub use crate::types::error::_access_denied_exception::AccessDeniedExceptionBuilder;

+#[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
 pub use crate::types::error::_authorization_pending_exception::AuthorizationPendingExceptionBuilder;

+#[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
 pub use crate::types::error::_expired_token_exception::ExpiredTokenExceptionBuilder;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_register_client",
+    feature = "op_start_device_authorization"
+))]
 pub use crate::types::error::_internal_server_exception::InternalServerExceptionBuilder;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_start_device_authorization"
+))]
 pub use crate::types::error::_invalid_client_exception::InvalidClientExceptionBuilder;

+#[cfg(any(feature = "op_create_token", feature = "op_create_token_with_iam"))]
 pub use crate::types::error::_invalid_grant_exception::InvalidGrantExceptionBuilder;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_register_client",
+    feature = "op_start_device_authorization"
+))]
 pub use crate::types::error::_invalid_request_exception::InvalidRequestExceptionBuilder;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_register_client"
+))]
 pub use crate::types::error::_invalid_scope_exception::InvalidScopeExceptionBuilder;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_register_client",
+    feature = "op_start_device_authorization"
+))]
 pub use crate::types::error::_slow_down_exception::SlowDownExceptionBuilder;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_start_device_authorization"
+))]
 pub use crate::types::error::_unauthorized_client_exception::UnauthorizedClientExceptionBuilder;

+#[cfg(any(
+    feature = "op_create_token",
+    feature = "op_create_token_with_iam",
+    feature = "op_register_client"
+))]
 pub use crate::types::error::_unsupported_grant_type_exception::UnsupportedGrantTypeExceptionBuilder;

+#[cfg(feature = "op_create_token_with_iam")]
 pub use crate::types::error::_invalid_request_region_exception::InvalidRequestRegionExceptionBuilder;

+#[cfg(feature = "op_register_client")]
 pub use crate::types::error::_invalid_client_metadata_exception::InvalidClientMetadataExceptionBuilder;

+#[cfg(feature = "op_register_client")]
 pub use crate::types::error::_invalid_redirect_uri_exception::InvalidRedirectUriExceptionBuilder;
